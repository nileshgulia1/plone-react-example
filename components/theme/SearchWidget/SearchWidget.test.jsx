import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';

import SearchWidget from './SearchWidget';

const mockStore = configureStore();

describe('SearchWidget', () => {
  it('renders a search widget component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <SearchWidget pathname="/blog" />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
