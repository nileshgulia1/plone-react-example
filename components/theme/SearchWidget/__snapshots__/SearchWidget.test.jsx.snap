// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchWidget renders a search widget component 1`] = `
<form
  action="/search"
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field searchbox"
  >
    <div
      className="ui transparent action input"
      title="Search"
    >
      <input
        disabled={undefined}
        name="SearchableText"
        onChange={[Function]}
        placeholder="Search Site"
        tabIndex={undefined}
        type="text"
        value=""
      />
      <button
        className="ui icon button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="search icon"
        />
      </button>
    </div>
  </div>
</form>
`;
